src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/GeneratorTests.Baselines.cs

```diff
@@ -965,5 +965,153 @@ public abstract class AbstractType_CannotInit
 
             Assert.Equal(2, result.Diagnostics.Where(diag => diag.Id == Diagnostics.TypeNotSupported.Id).Count());
         }
+
+        [Fact]
+        public async Task UnsupportedTypes()
+        {
+            string source = """
+                using System;
+                using System.Collections.Generic;
+                using System.Reflection;
+                using System.Runtime.Serialization;
+                using Microsoft.Extensions.Configuration;
+                using Microsoft.Extensions.DependencyInjection;
+
+                public class Program
+                {
+                	public static void Main()
+                	{
+                		ConfigurationBuilder configurationBuilder = new();
+                		IConfiguration configuration = configurationBuilder.Build();
+
+                        // var ws = config.GetSection("MySection");
+                        var w = new Options();
+                        configuration.Bind(w);
+                        configuration.Get<Options>(_ => { });
+
+                        // Should generate a diagnostics
+                        configuration.Get<List<IntPtr>>(_ => { });
```

Comment:
Turning this into a theory would let us test for other root type as well:
```suggestion
                        configuration.Get<{{unsupportedRootType}}>(_ => { });
```
Note that this would require `$$"""` syntax at the start of your triple-quote string.

Reply:
It is a good idea. The drawback of this will make the test much slower as we compare the result against the baseline.

Reply:
For root testing, I can split it to another test and don't have to compare to baseline as I want to check the diagnostics for this case.